components/trial-bar/button.jsx

```diff
@@ -0,0 +1,188 @@
+		PropTypes.array,
+		PropTypes.object,
+		PropTypes.string,
+	]),
+	/**
+	 * Name of the icon. Visit <a href="http://www.lightningdesignsystem.com/resources/icons">Lightning Design System Icons</a> to reference icon names.
+	 */
+	iconName: PropTypes.string,
+	/**
+	 * Path to the icon. This will override any global icon settings.
+	 */
+	iconPath: PropTypes.string,
+	/**
+	 * If omitted, icon position is centered.
+	 */
+	iconPosition: PropTypes.oneOf(['left', 'right']),
+	/**
+	 * Determines the size of the icon.
+	 */
+	iconSize: PropTypes.oneOf(['x-small', 'small', 'medium', 'large']),
+	/**
+	 * For icon variants, please reference <a href="http://www.lightningdesignsystem.com/components/buttons/#icon">Lightning Design System Icons</a>.
+	 */
+	iconVariant: PropTypes.oneOf([
+		'bare',
+		'container',
+		'border',
+		'border-filled',
+		'brand',
+		'more',
+		'global-header',
+	]),
+	/**
+	 * Id string applied to button node.
+	 */
+	id: PropTypes.string,
+	/**
+	 * If true, button/icon is white. Meant for buttons or utility icons on dark backgrounds.
+	 */
+	inverse: PropTypes.bool,
+	/**
+	 * Visible label on the button. If the button is an icon button with no label, you must use the <code>assistiveText.icon</code> prop.
+	 */
+	label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+	/**
+	 * Triggered when focus is removed.
+	 */
+	onBlur: PropTypes.func,
+	/**
+	 * Triggered when the button is clicked.
+	 */
+	onClick: PropTypes.func,
+	/**
+	 * Triggered when component is focused.
+	 */
+	onFocus: PropTypes.func,
+	/**
+	 * Triggered when a key is pressed down
+	 */
+	onKeyDown: PropTypes.func,
+	/**
+	 * Triggered when a key is pressed and released
+	 */
+	onKeyPress: PropTypes.func,
+	/**
+	 * Triggered when a key is released
+	 */
+	onKeyUp: PropTypes.func,
+	/**
+	 * Triggered when a mouse button is pressed down
+	 */
+	onMouseDown: PropTypes.func,
+	/**
+	 * Triggered when a mouse arrow hovers
+	 */
+	onMouseEnter: PropTypes.func,
+	/**
+	 * Triggered when a mouse arrow no longer hovers
+	 */
+	onMouseLeave: PropTypes.func,
+	/**
+	 * Triggered when a mouse button is released
+	 */
+	onMouseUp: PropTypes.func,
+	/**
+	 * If true, button scales to 100% width on small form factors.
+	 */
+	responsive: PropTypes.bool,
+	/**
+	 * Write <code>"-1"</code> if you don't want the user to tab to the button.
+	 */
+	tabIndex: PropTypes.string,
+	/**
+	 * Button type
+	 */
+	type: PropTypes.oneOf(['reset', 'submit', 'button']),
+	/**
+	 * HTML title attribute
+	 */
+	title: PropTypes.string,
+	/**
+	 * [Deprecated] Tooltip on button. Button should be a child of `Tooltip` instead.
+	 */
+	tooltip: PropTypes.node,
+	/**
+	 * Different types of buttons
+	 */
+	variant: PropTypes.oneOf([
+		'base',
+		'link',
+		'neutral',
+		'brand',
+		'outline-brand',
+		'destructive',
+		'success',
+		'text-destructive',
+		'icon',
+	]),
+	/**
+	 * Custom styles to be passed to the component
+	 */
+	style: PropTypes.object,
+};
+
```

Comment:
@garygong @davidlygagnon  I would recommend non-custom props for TrailBar's `Button` and `Dropdown` link to the Button and Dropdown doc pages in the comment on this line inside a comment which would be the larger type on the component description. This would allow the removal of the prop table from the docs and decrease duplicate lines of code which may not be updated in the future, for instance, when we add props to dropdown.

Something closer to `Radio` on this page https://react.lightningdesignsystem.com/components/radio-groups/

Source: https://github.com/salesforce/design-system-react/blob/master/components/radio-group/index.jsx#L4